src/compile/axis.ts

```diff
@@ -177,22 +178,32 @@ export namespace properties {
       }, labelsSpec);
     }
 
+    let textTemplate = '{{ datum.data }}';
+
     if (contains([NOMINAL, ORDINAL], fieldDef.type) && axis.labelMaxLength) {
       // TODO replace this with Vega's labelMaxLength once it is introduced
-      labelsSpec = extend({
-        text: {
-          template: '{{ datum.data | truncate:' + axis.labelMaxLength + '}}'
-        }
-      }, labelsSpec || {});
+      textTemplate = '{{ datum.data | truncate:' + axis.labelMaxLength + '}}';
```

Comment:
As mentioned in a slack conversation, Vega might use not use text template but rather calling formatting function directly.    

First, take a look why I think that to be the case.  :)